cms/src/test/java/uk/nhs/digital/externalstorage/workflow/externalFilePublish/ExternalFilePublishTaskTest.java

```diff
@@ -30,33 +41,163 @@
 
     private ExternalFilePublishTask externalFilePublishTask;
     private Session session;
-
     private Node rootNode;
+    private Node testFolder;
+    private final String testExternalReference = "ab/1c2d3e/one.pdf";
 
     @Before
     public void setUp() throws RepositoryException {
         initMocks(this);
         given(s3Connector.publishResource(any(String.class)))
             .willReturn(true);
+        given(s3Connector.unpublishResource(any(String.class)))
+            .willReturn(true);
 
         Repository repository = MockJcr.newRepository();
         session = repository.login();
         rootNode = session.getRootNode();
+        testFolder = rootNode.addNode("lorem-ipsum", NT_FOLDER);
         externalFilePublishTask = new ExternalFilePublishTask();
 
         given(workflowContext.getInternalWorkflowSession()).willReturn(session);
         externalFilePublishTask.setWorkflowContext(workflowContext);
     }
 
     @Test
-    public void shouldCallS3() throws WorkflowException, RepositoryException {
-        Node docNode = rootNode.addNode("test-document");
-        addVariantNode(docNode, UNPUBLISHED);
-        MockJcr.setQueryResult(session, getQueryResults());
+    public void shouldPublishResources() throws WorkflowException, RepositoryException {
```

Comment:
The name could use indicating that this test method is about ensuring that external resources of documents other than publications without datasets will be published.

Reply:
will rename to `shouldPublishAnyDocumentResources`